component/main.jsonnet

```diff
@@ -177,14 +177,43 @@ local plan = [
     ) else
       [];
 
+  local command(p) =
+    if std.objectHas(p, 'command') then (
+      if std.type(p.command) == 'string' then (
+        [ p.command ]
+      )
+      else (
+        if std.type(p.command) == 'array' then (
+          p.command
+        ) else
+          error 'Field `command` of plan "%(name)s" is not an array nor a string' % p
+      )
+    ) else
+      [];
```

Comment:
What happens when a Plan with an empty `command` array is created? Is the command from the image used? Does the System Upgrade Controller accept such a Plan?

It may make more sense to raise an error if a plan without command is defined

Reply:
I'll check this, thx for the input

Reply:
an empty command array results in using the command from the imge and the plan is accepted.
But I could also just not set the command field if an empty array is given. The example in https://github.com/rancher/system-upgrade-controller/blob/master/examples/k3s-upgrade.yaml also does not have the command set, so this is valid